providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelConnectorTest.java

```diff
@@ -57,12 +58,16 @@ void setUp() throws Exception {
     }
 
     private void setupTestGrpcServer() throws IOException {
+        var testSocket = new ServerSocket(0);
+        var port = testSocket.getLocalPort();
```

Comment:
```suggestion
        var port = new Random().nextInt(49152, 65535);
```
If we just want a random port number, why not use a random number?

Reply:
because the port can still be allocated by a service, we don't know what is running on the host, and `ServerSocket(0)` will pick an un-allocated port.